apps/system/lockscreen/js/lockscreen_state_manager.js

```diff
@@ -415,10 +416,52 @@
         this.onUnlockingApp();
         break;
       case 'lockscreenslide-activate-right':
+        isFromNotification = false;
+        appID = '';
+        this.onActivateUnlock();
+        break;
       case 'lockscreen-notification-request-activate-unlock':
         this.onActivateUnlock();
+        isFromNotification = true;
+        appID = detail.id;
         break;
       case 'lockscreen-request-unlock':
+      // if app is launched from the lockscreen notification screen then it will
+      // find the that app in system and launch it.
+        if (isFromNotification) {
+          isFromNotification = false;
+          var request = navigator.mozApps.mgmt.getAll();
```

Comment:
I don't think any of this code is necessary. The lockscreen can already launch apps when the user clicks on a notification. That is part of now notifications work. You should not have to turn the lockscreen into a window manager in order to respond to launch an app.  I'm not sure where the existing code is, but I suspect it is related to the tap() method at apps/system/js/notification.js:312